rdkit/Chem/Fingerprints/ClusterMols.py

```diff
@@ -39,22 +39,23 @@ def GetDistanceMatrix(data, metric, isSimilarity=1):
 
   """
   nPts = len(data)
-  res = numpy.zeros((nPts * (nPts - 1) // 2), numpy.float)
+  distsMatrix = [0.0] * (nPts * (nPts - 1) // 2)
   nSoFar = 0
   for col in range(1, nPts):
+    fp1 = data[col][1]
     for row in range(col):
-      fp1 = data[col][1]
       fp2 = data[row][1]
-      if fp1.GetNumBits() > fp2.GetNumBits():
-        fp1 = DataStructs.FoldFingerprint(fp1, fp1.GetNumBits() / fp2.GetNumBits())
-      elif fp2.GetNumBits() > fp1.GetNumBits():
-        fp2 = DataStructs.FoldFingerprint(fp2, fp2.GetNumBits() / fp1.GetNumBits())
-      sim = metric(fp1, fp2)
+      nBits1, nBits2 = fp1.GetNumBits(), fp2.GetNumBits()
```

Comment:
Move the assignment to nBits1 out of the loop as well.

Reply:
OK, I got your idea! I would fix it in the next patch. Thank you for your review.